src/shared/read-agents.c

```diff
@@ -596,6 +596,10 @@ void delete_sqlite(const char *id, const char *name)
 /* Delete diff folders */
 void delete_diff(const char *name)
 {
+    if (name == NULL || *name == '\0') {
+        return;
+    }
+
```

Comment:
I'm not to fond on the idea of changing a generic function to include a special condition, can we move the check and catch it [here](https://github.com/wazuh/wazuh/blob/1a08a3b47b7dcb2a460bbf39ef766d81b8865447/src/wazuh_modules/wm_database.c#L333-L335)?